pkg/controller/hostpathprovisioner/controller.go

```diff
@@ -419,8 +424,13 @@ func (r *ReconcileHostPathProvisioner) checkDegraded(logger logr.Logger, cr *hos
 	if err != nil {
 		return true, err
 	}
+	daemonSetCsi := &appsv1.DaemonSet{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Name: fmt.Sprintf("%s-csi", MultiPurposeHostPathProvisionerName), Namespace: namespace}, daemonSet)
+	if err != nil {
+		return true, err
+	}
 
-	if !checkDaemonSetReady(daemonSet) {
+	if !checkDaemonSetReady(daemonSet) || checkDaemonSetReady(daemonSetCsi) {
```

Comment:
`!checkDaemonSetReady(daemonSet) || checkDaemonSetReady(daemonSetCsi) `
non csi false and csi true means degraded? shouldn't be 
`!checkDaemonSetReady(daemonSet) || !checkDaemonSetReady(daemonSetCsi) `

Reply:
Yes, so it turns out I was looking up the wrong daemonset right above. so it should be correct now.